packages/dds/merge-tree/src/mergeTreeNodes.ts

```diff
@@ -397,6 +397,9 @@ export class MergeBlock implements IMergeNodeCommon {
 	public ordinal: string = "";
 	public cachedLength: number | undefined = 0;
 
+	public rightmostTiles: Readonly<MapLike<Marker>>;
```

Comment:
We should doc these fields as it'll make semantics of `blockUpdate` clearer. Perhaps something like:

"Maps each tile label in this block to the rightmost (i.e. furthest) marker associated with that tile label. When combined with the tree structure of `MergeBlock`s, this allows accelerated queries for nearest tile with a certain label before a given position."